gdal/frmts/netcdf/netcdfsg.cpp

```diff
@@ -0,0 +1,1047 @@
+			{
+				// not a geometry variable, continue
+				continue;
+			}
+
+			// Also if present but empty, go on
+			if(contname2_len == 0) continue;
+
+			// Otherwise, geometry: see what container it has
+			buf = new char[contname2_len];
+			if(nc_get_att_text(this->nc, curr, CF_SG_GEOMETRY, buf)!= NC_NOERR)
+			{
+				delete[] buf;
+				continue;
+			}
+
+			// If matches, then establish a reference by placing this variable's data in both vectors
+			if(!strcmp(contname, buf))
+			{
+				char property_name[NC_MAX_NAME];
+				nc_inq_varname(this->nc, curr, property_name);
+				
+				std::string n(property_name);
+				v_ids.push_back(curr);
+				v_headers.push_back(n);
+			}
+			
+			delete[] buf;
+		}
+	}
+
+	// Exception Class Implementations
+	SG_Exception_Dim_MM::SG_Exception_Dim_MM(char* container_name, const char* field_1, const char* field_2)
+	{
+		const char * format = "%s: One or more dimensions of \"%s\" and \"%s\" must match but do not match.";
+		// This may allocate a couple more than needed, potential improvement: optimize a couple of bytes later
+		size_t s = strlen(field_1) + strlen(field_2) + strlen(format) + strlen(container_name) + 1;
+		err_msg = new char[s];
+		snprintf(err_msg, s, format, container_name, field_1, field_2);
+	}
+
+	SG_Exception_Existential::SG_Exception_Existential(char* container_name, const char* missing_name)
+	{
+		const char * format = "[%s] The property or the variable associated with \"%s\" is missing.";
+
+		// This may allocate a couple more than needed, potential improvement: optimize a couple of bytes later
+		size_t s = strlen(missing_name) + strlen(format) + strlen(container_name) + 1;
+		err_msg = new char[s];
+		snprintf(err_msg, s, format, container_name, missing_name);
+	}
+
+	SG_Exception_Dep::SG_Exception_Dep(char* container_name, const char* arg1, const char* arg2)
+	{
+		const char * format = "[%s] The attribute \"%s\" may not exist without the attribute \"%s\" existing.";
+		
+		// This may allocate a couple more than needed, potential improvement: optimize a couple of bytes later
+		size_t s = strlen(arg1) + strlen(arg2) + strlen(format) + strlen(container_name) + 1;
+		err_msg = new char[s];
+		snprintf(err_msg, s, format, container_name, arg1, arg2);
+	}
+	
+	SG_Exception_BadSum::SG_Exception_BadSum(char* container_name, const char* arg1, const char* arg2)
+	{
+		const char * format = "[%s] The sum of all values in \"%s\" and \"%s\" do not match.";
+		
+		// This may allocate a couple more than needed, potential improvement: optimize a couple of bytes later
+		size_t s = strlen(arg1) + strlen(arg2) + strlen(format) + strlen(container_name) + 1;
+		err_msg = new char[s];
+		snprintf(err_msg, s, format, container_name, arg1, arg2);
+	}
+
+	// to get past linker
+	SG_Exception::~SG_Exception()
+	{
+
+	}
+
+	// Helpers
+	// following is a short hand for a clean up and exit, since goto isn't allowed
+	int getCFMinorVersion(int ncid)
```

Comment:
what about if CF 2.X is released ? Would't it make sense to return a double with the CF version and test against that ? And using sscanf(attrval, "CF-%f", &version) could simplify a lot this function.

Reply:
I'm not sure if we can say that CF-1.8 geometries will be usable in the CF-2.x convention, as 2.x would be a new major release potentially breaking this type of encoding. But sure I can change it so that it works with >= 1.8 CFs :+1: will fix